app/domain/authentication/common/extract_resource_restrictions.rb

```diff
@@ -0,0 +1,57 @@
+require 'command_class'
+
+module Authentication
+  module Common
+
+    ExtractResourceRestrictions = CommandClass.new(
+        dependencies: {
+            role_class:     ::Role,
+            resource_class: ::Resource,
+            logger:         Rails.logger
+        },
+        inputs:   %i(authenticator_name service_id host_name account)
+    ) do
+
+      def call
+        fetch_all_role_annotations
+        convert_to_resource_restrictions
+        @resource_restrictions
+      end
+
+      private
+
+      def fetch_all_role_annotations
+        # Validate role exists, otherwise getting role annotations return empty hash.
+        raise Errors::Authentication::Security::RoleNotFound, role_id unless @resource_class[role_id]
```

Comment:
how about moving this to a new method `validate_role_exists` and call it in the `call` method?

Reply:
I did at first, but @eranha noted out that this command class should not be responsible to validate it.
I kept it here with a comment to explain why it is still needed.

Reply:
you can still do it but move it to another method and have the comment above that method

Reply:
just so we dont have any implicit background operations